superset/db_engine_specs/base.py

```diff
@@ -909,19 +909,22 @@ def modify_url_for_impersonation(
             url.username = username
 
     @classmethod
-    def get_configuration_for_impersonation(  # pylint: disable=invalid-name
-        cls, uri: str, impersonate_user: bool, username: Optional[str]
-    ) -> Dict[str, str]:
+    def update_connect_args_for_impersonation(
+        cls,
+        connect_args: Dict[Any, Any],
+        uri: str,
+        impersonate_user: bool,
```

Comment:
I would remove this `bool` from the signature and rather check it in `core.py` prior to calling this method (no point in calling the method if we're not impersonating).